src/train_cmd.cpp

```diff
@@ -1872,9 +1851,9 @@ void ReverseTrainDirection(Train *v)
 		return;
 	}
 
-	/* TrainExitDir does not always produce the desired dir for depots and
+	/* VehicleExitDir does not always produce the desired dir for depots and
 	 * tunnels/bridges that is needed for UpdateSignalsOnSegment. */
-	DiagDirection dir = TrainExitDir(v->direction, v->track);
+	DiagDirection dir = VehicleExitDir(v->direction, v->track);
```

Comment:
An alternative:
```
DiagDirection dir;
if (depot or tunnel) {
   dir = invalid dir
} else {
   calculate vehicle exit dir
}
```

Reply:
I wasn't planning on touching this code, just the function was renamed.

Reply:
I just spotted it. Maybe someone thinks this is really important and tries to write a patch for it. But this is so stupid no one will. :P